### PR TITLE
ibmcloud: Included Readiness Probe along with nginx containers

### PR DESCRIPTION
**Pull request by sudharshanibm3**
Included Readiness Probe along with nginx containers

Fixes:  #1450

### CODE REVIEW

**Comment by mkulke** [comment]
Why would the dockerhub's nginx image - arguably the most testes oci image - require an init container?

File: test/e2e/ibmcloud_test.go [hunk 1]
```diff
@@ -308,6 +308,7 @@ func newPodWithPVCFromIBMVPCBlockDriver(namespace, podName, containerName, image
 					Name:            containerName,
 					Image:           imageName,
 					ImagePullPolicy: corev1.PullAlways,
+					Ports:           []corev1.ContainerPort{{ContainerPort: 80}},
```
Comment (mkulke) [inline test/e2e/ibmcloud_test.go:311]: Should we add that to all test suites?

Comment (sudharshanibm3) [reply test/e2e/ibmcloud_test.go:311]: No I think, I added ContainerPort to test cases which only executes commands inside container

Comment (mkulke) [reply test/e2e/ibmcloud_test.go:311]: I mean not just for IBM, but make this the default everywhere.

Comment (sudharshanibm3) [reply test/e2e/ibmcloud_test.go:312]: Hi @mkulke ,
Sorry I updated the code now
- Here I added `containerport` and `ReadinessProbe` manually for ibm specific test cases - [Ref](https://github.com/confidential-containers/cloud-api-adaptor/pull/1502/files#diff-96d31bf66f849bfcdfdd93f021bd2de1bf27e2d9cf6ceb7e79f2afceef4fc3a7R312-R322)
- But for other test cases created a common function (withContainerPort) to add `containerport` and `ReadinessProbe` in [common.go](https://github.com/confidential-containers/cloud-api-adaptor/pull/1502/files#diff-2c7fc9c9c12632c5e550abeb31614356abd17710d948a8664f9d2f2163e38b0dR22-R38)

Comment (mkulke) [reply test/e2e/ibmcloud_test.go:312]: Ok, I guess we can add it to other clouds & libvirt in separate PRs. I was just concerned to introduce different test behavior in the individual backends.

cc @kartikjoshi21 @wainersm

**Review by mkulke — APPROVED** [review]

File: test/e2e/common.go [hunk 2]
```diff
@@ -40,6 +58,19 @@ func withImagePullSecrets(secretName string) podOption {
...
+// 		p.Spec.InitContainers = []corev1.Container{
+// 			{
+// 				Name:    "useradd-init-container",
+// 				Image:   "busybox",
+// 				Command: []string{"adduser", "-D", "nginx"},
+// 			},
+// 		}
+// 	}
+// }
+
```
Comment (stevenhorsman) [inline test/e2e/common.go:73]: Can we remove the commented out code from this PR please as we aren't using it.

**Review by stevenhorsman — APPROVED** [review]
LGTM. I agree with Magnus that adding the probe to the other cloud providers makes sense, but we can start with it here and monitor the results first. Thanks!

**Pull request merged**